csrc/cpu_ops.cpp

```diff
@@ -55,19 +57,18 @@ void quantize_cpu(float *code, float *A, float *absmax, unsigned char *out, long
           arg->threadidx = block_idx / blocksize;
           arg->blocksize = blocksize;
 
-          pthread_create(&threads[chunks_processed], NULL, &quantize_block, (void *) arg);
+          wave_storage.emplace_back(std::async(std::launch::async, [arg] {quantize_block(arg); }));
           chunks_processed += 1;
           if(chunks_processed == valid_chunks){ break; }
       }
 
-      for (int i = 0; i < valid_chunks; i++)
-          int err = pthread_join(threads[i], NULL);
+      for (int i = 0; i < wave_storage.size(); i++)
+          wave_storage[i].wait();
```

Comment:
This is inefficient. If we want to keep 256 wave occupancy, as soon as one of the futures is done, we need to queue a new one. Alternatively we could queue everything from the start :) as we're not going to bomb the OS.

Hopefully the libCxx implementation is doing the right thing and using thread pools. I'll need to check if I don't need to do something special in the async launch.